### PR TITLE
[MM-20796] Un-reverted the Ctrl+Tab fix and also made sure the tabs go in the right order

### PR DESCRIPTION
**Pull request by devinbinnie**
Before submitting, please confirm you've
 - [x] read and understood our [Contributing Guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)
 - [x] completed [Mattermost Contributor Agreement](http://www.mattermost.org/mattermost-contributor-agreement/)
 - [x] executed `npm run lint:js` for proper code formatting

Please provide the following information:

**Summary**
The fix in this PR: https://github.com/mattermost/desktop/pull/1127
Was overwritten by this PR: https://github.com/mattermost/desktop/pull/1056
This PR puts the fix back.

Also found a bug where tabbing through when you've re-ordered the tabs would tab through in the wrong order, also fixed.

**Issue link**
https://mattermost.atlassian.net/browse/MM-20796

### CODE REVIEW

File: src/browser/components/MainPage.jsx [hunk 1]
```diff
@@ -156,10 +156,24 @@ export default class MainPage extends React.Component {
       this.handleSelect(key);
     });
     ipcRenderer.on('select-next-tab', () => {
-      this.handleSelect(this.state.key + 1);
+      const currentOrder = this.props.teams[this.state.key].order;
+      const nextIndex = this.props.teams.map((team, index) => {
+        return {
+          index,
+          order: (team.order - currentOrder) - 1,
```
Comment (Willyfrog) [inline src/browser/components/MainPage.jsx]: logic here can be made more readable if we swap the parens
```suggestion
          order: team.order - (currentOrder + 1),
```
which looks more like "next tab"

File: src/browser/components/MainPage.jsx [hunk 2]
```diff
@@ -156,10 +156,24 @@ export default class MainPage extends React.Component {
...
+        };
+      }).find((team) => team.order === 0 || Math.abs(team.order) === this.props.teams.length);
```
Comment (Willyfrog) [inline src/browser/components/MainPage.jsx]: I wonder if it wouldn't be easier if we calculate the next one and then just look for it.
```js
const nextOrder = (currentOrder + 1) % this.props.teams.length;
const nextIndex = this.props.teams.find((team) => team.order === nextOrder;
```

File: src/browser/components/MainPage.jsx [hunk 3]
```diff
@@ -156,10 +156,24 @@ export default class MainPage extends React.Component {
...
     ipcRenderer.on('select-previous-tab', () => {
-      this.handleSelect(this.state.key - 1);
+      const currentOrder = this.props.teams[this.state.key].order;
+      const nextIndex = this.props.teams.map((team, index) => {
+        return {
+          index,
+          order: (team.order - currentOrder) + 1,
+        };
+      }).find((team) => team.order === 0 || Math.abs(team.order) === this.props.teams.length);
+      this.handleSelect(nextIndex.index);
```
Comment (Willyfrog) [inline src/browser/components/MainPage.jsx]: same comments as above, but going back ;)

File: src/browser/components/MattermostView.jsx [hunk 4]
```diff
@@ -335,7 +335,7 @@ export default class MattermostView extends React.Component {
     if (this.props.withTab) {
       classNames.push('mattermostView-with-tab');
     }
-    if (!this.props.active || this.state.errorInfo) {
+    if (!this.props.active) {
```
Comment (Willyfrog) [inline src/browser/components/MattermostView.jsx]: not sure of why this change is here :/ seems unrelated to the PR, can you help me understand?

Comment (devinbinnie) [reply src/browser/components/MattermostView.jsx]: See this PR: https://github.com/mattermost/desktop/pull/1127

File: src/browser/components/MainPage.jsx (hunk 2 above)
Comment (devinbinnie) [reply src/browser/components/MainPage.jsx]: Order isn't 0 based, so this doesn't work.

File: src/browser/components/MainPage.jsx [hunk 5]
```diff
@@ -156,10 +156,24 @@ export default class MainPage extends React.Component {
       this.handleSelect(key);
     });
     ipcRenderer.on('select-next-tab', () => {
-      this.handleSelect(this.state.key + 1);
+      const currentOrder = this.props.teams[this.state.key].order;
+      const nextIndex = this.props.teams.map((team, index) => {
+        return {
+          index,
+          order: team.order - (currentOrder - 1),
```
Comment (Willyfrog) [inline src/browser/components/MainPage.jsx]:
```suggestion
          order: team.order - (currentOrder + 1),
```
now it's going backwards

File: src/browser/components/MainPage.jsx [hunk 6]
```diff
@@ -156,10 +156,24 @@ export default class MainPage extends React.Component {
...
+      }).find((team) => team.order === 0 || Math.abs(team.order) === this.props.teams.length);
+      this.handleSelect(nextIndex.index);
     });
     ipcRenderer.on('select-previous-tab', () => {
-      this.handleSelect(this.state.key - 1);
+      const currentOrder = this.props.teams[this.state.key].order;
+      const nextIndex = this.props.teams.map((team, index) => {
+        return {
+          index,
+          order: team.order - (currentOrder + 1),
```
Comment (Willyfrog) [inline src/browser/components/MainPage.jsx]:
```suggestion
          order: team.order - (currentOrder - 1),
```

File: src/browser/components/MainPage.jsx (hunk 2 above)
Comment (Willyfrog) [reply src/browser/components/MainPage.jsx]: just checked and it is 0 based, if you think we are assuming it is not at some other line of code, we should revisit that part for possible bugs.

File: src/browser/components/MainPage.jsx (hunk 6 above)
Comment (devinbinnie) [reply src/browser/components/MainPage.jsx]: whoops...

**Comment by Willyfrog** [comment]
gah! hit the button too soon, there is an alternate implementation:
```js
    ipcRenderer.on('select-next-tab', () => {
      const currentOrder = this.props.teams[this.state.key].order;
      const nextOrder = ((currentOrder + 1) % this.props.teams.length);
      const nextIndex = this.props.teams.findIndex((team) => team.order === nextOrder);
      this.handleSelect(nextIndex);
    });

    ipcRenderer.on('select-previous-tab', () => {
      const currentOrder = this.props.teams[this.state.key].order;

      // js modulo operator returns a negative number if result is negative, so we have to ensure it's positive
      const nextOrder = ((this.props.teams.length + (currentOrder - 1)) % this.props.teams.length);
      const nextIndex = this.props.teams.findIndex((team) => team.order === nextOrder);
      this.handleSelect(nextIndex);
    });
```
